clients/src/main/java/org/apache/kafka/clients/admin/DescribeReplicaLogDirResult.java

```diff
@@ -0,0 +1,110 @@
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Collection;
+import java.util.concurrent.ExecutionException;
+import org.apache.kafka.common.requests.DescribeLogDirsResponse;
+
+
+/**
+ * The result of {@link AdminClient#describeReplicaLogDir(Collection)}.
+ *
+ * The API of this class is evolving, see {@link AdminClient} for details.
+ */
+@InterfaceStability.Evolving
+public class DescribeReplicaLogDirResult {
+    private final Map<TopicPartitionReplica, KafkaFuture<ReplicaLogDirInfo>> futures;
+
+    DescribeReplicaLogDirResult(Map<TopicPartitionReplica, KafkaFuture<ReplicaLogDirInfo>> futures) {
+        this.futures = futures;
+    }
+
+    /**
+     * Return a map from replica to future which can be used to check the log directory information of individual replicas
+     */
+    public Map<TopicPartitionReplica, KafkaFuture<ReplicaLogDirInfo>> values() {
+        return futures;
+    }
+
+    /**
+     * Return a future which succeeds if log directory information of all replicas are available
+     */
+    public KafkaFuture<Map<TopicPartitionReplica, ReplicaLogDirInfo>> all() {
+        return KafkaFuture.allOf(futures.values().toArray(new KafkaFuture[0])).
+            thenApply(new KafkaFuture.Function<Void, Map<TopicPartitionReplica, ReplicaLogDirInfo>>() {
+                @Override
+                public Map<TopicPartitionReplica, ReplicaLogDirInfo> apply(Void v) {
+                    Map<TopicPartitionReplica, ReplicaLogDirInfo> replicaLogDirInfos = new HashMap<>();
+                    for (Map.Entry<TopicPartitionReplica, KafkaFuture<ReplicaLogDirInfo>> entry : futures.entrySet()) {
+                        try {
+                            replicaLogDirInfos.put(entry.getKey(), entry.getValue().get());
+                        } catch (InterruptedException | ExecutionException e) {
+                            // This should be unreachable, because allOf ensured that all the futures completed successfully.
+                            throw new RuntimeException(e);
+                        }
+                    }
+                    return replicaLogDirInfos;
+                }
+            });
+    }
+
+    static public class ReplicaLogDirInfo {
+        // The log directory of the primary replica of this partition on the given broker.
+        // Null if the primary replica is not found for this partition on the given broker.
+        public final String currentReplicaLogDir;
+        // The log directory of the temporary replica of this partition on the given broker.
+        // Null if the temporary replica is not found for this partition on the given broker.
+        public final String temporaryReplicaLogDir;
+        // The LEO of the primary replica  - LEO of the temporary replica.
+        // -1 if either primary or the temporary replica is not found for this partition on the given broker.
+        public final long temporaryReplicaOffsetLag;
```

Comment:
The class `ReplicaLogDirInfo` doesn't include that currentReplicaLag because this information is not currently used in this patch. Previously I think we can include `currentReplicaLag` in this when it is needed, i.e. to track reassignment progress in KIP-179. I have updated the patch to include `currentReplicaLag` in `ReplicaLogDirInfo`.

I am not sure I fully understand the second question. What information should we add or remove from `DescribeLogDirsResult`? I think `DescribeLogDirsResult` and `DescribeReplicaLogDirResult` has different format because they are used to serve two different use-cases. These two classes are structured in a way that makes the respective AdminClient APIs easier to use.